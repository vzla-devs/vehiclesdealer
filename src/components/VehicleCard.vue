<template>
  <v-card>
    <v-img :src="imageUrl" />
    <v-card-title primary-title>
      <div>
        <h3 class="headline">{{ brand }}</h3>
        <div> {{ year }} </div>
      </div>
    </v-card-title>
  </v-card>
</template>

<script>
export default {
  props: {
    brand: String,
    model: String,
    year: String,
    imageUrl: String
  }
}
</script>


<style lang="scss" scoped>
.card {
  border-radius: $border-radius;
  background-color: lightgray;
}
</style>
