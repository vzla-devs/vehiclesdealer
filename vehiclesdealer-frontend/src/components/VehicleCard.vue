<template>
  <div>
    <v-card
      hover
      class="card"
      @click="onClick"
    >
      <v-img
        :src="image"
      />
      <v-card-title primary-title>
        <div class="description-container">
          <h3 class="headline description">
            {{ formattedTitle }}
          </h3>
          <div class="price">
            {{ formattedPrice }}
          </div>
        </div>
      </v-card-title>
    </v-card>
  </div>
</template>

<script>
export default {
  props: {
    image: { type: String, required: true },
    title: { type: String, required: true },
    price: { type: Number, required: true }
  },
  computed: {
    formattedTitle () { return this.title.toUpperCase() },
    formattedPrice () { return `${this.price} â‚¬` }
  },
  methods: {
    onClick () {
      this.$emit('onClick')
    }
  }
}
</script>

<style lang="scss" scoped>
.card {
  display: grid;
}
.description-container {
  width: 100%;
}
.description {
  text-align: justify;
  font-size: 24px;
  font-weight: 400;
}
.price {
  text-align: end;
  font-size: 16px;
  font-weight: 500;
}
</style>
