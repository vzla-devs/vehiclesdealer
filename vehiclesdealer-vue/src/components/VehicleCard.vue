<template>
  <v-card
    :hover="true"
    class="card"
  >
    <v-img :src="imageUrl" />
    <v-card-title primary-title>
      <div class="description-container">
        <h3 class="headline description">
          {{ formattedDescription }}
        </h3>
        <div class="price">
          {{ formattedPrice }}
        </div>
      </div>
    </v-card-title>
  </v-card>
</template>

<script>
export default {
  props: {
    brand: { type: String, required: true },
    model: { type: String, required: true },
    year: { type: Number, required: true },
    price: { type: Number, required: true },
    imageUrl: { type: String, required: true }
  },
  computed: {
    formattedDescription () { return `${this.brand} ${this.model} - ${this.year}` },
    formattedPrice () { return `${this.price} â‚¬` }
  }
}
</script>

<style lang="scss" scoped>
.card {
  border-radius: $border-radius;
}
.description-container {
  width: 100%;
}
.description {
  text-transform: uppercase;
  text-align: justify;
  font-size: 24px;
  font-weight: 400;
}
.price {
  text-align: end;
  font-size: 16px;
  font-weight: 500;
}
</style>
