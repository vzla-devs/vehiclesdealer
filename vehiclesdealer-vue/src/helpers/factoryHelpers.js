import { shallowMount } from '@vue/test-utils'
import Vuex from 'vuex'

export function wrapperBuilderFactory ({ component, localVue, stubs }) {
  let propsData = {}
  let store

  function withProps (props) {
    propsData = { ...props }
    return self
  }

  function withStore (newStore) {
    store = newStore
    return self
  }

  function build () {
    store = store ? new Vuex.Store({ ...store }) : undefined
    return shallowMount(component, { localVue, stubs, propsData, store })
  }

  const self = {
    withProps,
    withStore,
    build
  }
  return self
}