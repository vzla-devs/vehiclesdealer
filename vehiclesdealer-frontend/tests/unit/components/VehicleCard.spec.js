import VehicleCard from '@/components/VehicleCard.vue'
import ImageStub from '@tests/components/stubs/ImageStub.vue'
import { componentBuilder } from '@tests/helpers/builderHelpers'

describe('VehicleCard.vue', () => {
  it('renders correctly', () => {
    const brand = 'anyBrand'
    const model = 'anyModel'
    const year = 2019
    const price = 9999
    const imageUrl = 'anyUrl'

    const wrapper = aVehicleCard().withProps({ brand, model, year, price, imageUrl }).build()

    expect(wrapper.element).toMatchSnapshot()
  })

  function aVehicleCard () {
    const stubs = { 'v-img': ImageStub, 'v-card': true, 'v-card-title': true }
    return componentBuilder(VehicleCard).withStubs(stubs)
  }
})
