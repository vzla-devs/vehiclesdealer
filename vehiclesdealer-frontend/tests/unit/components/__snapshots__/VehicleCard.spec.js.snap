// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard.vue renders correctly 1`] = `
<div>
  <v-card-stub
    class="card"
    hover="true"
  >
    <img
      src="anyUrl"
    />
     
    <v-card-title-stub
      primary-title=""
    >
      <div
        class="description-container"
      >
        <h3
          class="headline description"
        >
          
          anyBrand anyModel - 2019
        
        </h3>
         
        <div
          class="price"
        >
          
          9999 â‚¬
        
        </div>
      </div>
    </v-card-title-stub>
  </v-card-stub>
</div>
`;
